hedgehog/src/Hedgehog/Internal/Property.hs

```diff
@@ -550,33 +561,52 @@ assert b = do
   else
     withFrozenCallStack failure
 
+-- | Fails the test and shows a git-like diff if the operation
+--   evaluates to 'False' when applied to its arguments.
+--
+diff :: (
+    MonadTest m
+  , Show a
+  , Show b
+  , HasCallStack
+  )
+  => a
+  -> (a -> b -> Bool)
+  -> b
+  -> m ()
+diff x op y = do
+  ok <- withFrozenCallStack $ eval (x `op` y)
+  if ok then
+    success
+  else
+    withFrozenCallStack $ failDiff x y
+
+comparison :: (MonadTest m, Show a, Show b, HasCallStack) => a -> (a -> b -> Bool) -> b -> m ()
+comparison = diff
+
+relation :: (MonadTest m, Show a, Show b, HasCallStack) => a -> (a -> b -> Bool) -> b -> m ()
+relation = diff
+
+relate :: (MonadTest m, Show a, Show b, HasCallStack) => a -> (a -> b -> Bool) -> b -> m ()
+relate = diff
+
```

Comment:
See comment at https://github.com/hedgehogqa/haskell-hedgehog/pull/196/files#r276932847.

Reply:
I think `diff` sounds more clear than all the others, short and sweet, *diff this with that* :rocket: :rocket: :rocket: